### PR TITLE
BLS signer tests

### PR DESCRIPTION
**Pull request by blakecduncan**
## What is this PR doing?

- Adding tests for a normal signer that interacts with an erc20 and an erc721 contract.
- There is a future issue that will swap out the normal signer with a BlsSigner to make sure everything works. (@JohnGuilding  not sure if you have the issue number?)

## How can these changes be manually tested?
- Tests pass

## Does this PR resolve or contribute to any issues?
https://github.com/web3well/bls-wallet/issues/385

## Checklist
- [x] I have manually tested these changes
- [x] Post a link to the PR in the group chat

## Guidelines
- If your PR is not ready, mark it as a draft
- The `resolve conversation` button is for reviewers, not authors
  - (But add a 'done' comment or similar)

### CODE REVIEW

File: contracts/package.json
Comment (blakecduncan) [inline contracts/package.json]: Is it okay if I bump this up?

Comment (jacque006) [reply contracts/package.json]: I don't see any changes that should impact us. @jzaki can give final say. 

Generally, in earlier stages of a project it is fine to bump this at will. Later on, it can be less desirable as:
- Many OpenZepplin updates have breaking changes, even in non-major version changes. This version jump [has 3](https://github.com/OpenZeppelin/openzeppelin-contracts/releases/tag/v4.8.0)
- Our contracts will be audited at a specific commit/tag with a specific OpenZepplin version used. Bumping this *could* require a new audit pass/tests.
- This will likely change our contract byte code, requiring new contract deploys and client lib updates.

File: contracts/clients-tests/Signer.test.ts
Comment (JohnGuilding) [inline contracts/clients-tests/Signer.test.ts]: Should this say `// Send mockERC20 to signer one` instead?

File: contracts/clients-tests/Signer.test.ts
Comment (JohnGuilding) [inline contracts/clients-tests/Signer.test.ts]: `signer.sendTransaction` is a well tested method now from the other provider work. So think it's safe to delete this whole test, and this file can just be concerned with testing the contract interactions

File: contracts/clients-tests/Signer.test.ts
Comment (JohnGuilding) [inline contracts/clients-tests/Signer.test.ts]: Thoughts on using `erc20ToTransfer ` here instead of doing a calculation with BigNumber? Same thought for L118 below

**Review by JohnGuilding — COMMENTED** [review]
LGTM - few really minor comments.

The issue number for the story to replace the signer with our bls signer is #414 and I'm working on it now. So can do that once this is merged in.

One question before approving, do all of the tests in your `clients-tests` directory pass? Some of mine are failing? I've ran the setup script, deleted artifacts, cache, & typechain types to no avail.

**Review by JohnGuilding — APPROVED** [review]
Looks good!

**Pull request merged**
